src/framework/components/collision/component.js

```diff
@@ -377,6 +404,41 @@ class CollisionComponent extends Component {
         }
     }
 
+
+    /**
+     * @returns {Vec3} The world position for the physics body taking into account of any offsets.
+     * @private
+     */
+    _getBodyPosition() {
+        const pos = this.entity.getPosition();
+        const rot = this.entity.getRotation();
+
+        if (this._hasOffset) {
+            const lo = this.data.linearOffset;
+            quat.copy(rot).transformVector(lo, vec3);
+            vec3.add((pos));
+            return vec3;
+        }
+
+        return pos;
+    }
+
+    /**
+     * @returns {Quat} The world rotation for the physics body taking into account of any angular offset.
+     * @private
+     */
+    _getBodyRotation() {
+        const rot = this.entity.getRotation();
+
+        if (this._hasOffset) {
+            const ao = this.data.angularOffset;
+            quat.copy(rot).mul(ao);
+            return quat;
+        }
+
+        return rot;
+    }
+
```

Comment:
Wondering if these should be public API as they would need to be used by the Editor for visualisation? Or just a comment that they will be used by the Editor be enough when it comes to Editor integration?

Reply:
It feels like it could be useful to have these public in general.

Reply:
Made public and change name from body to shape to be more inline with the rest of the API and terms